db.go

```diff
@@ -0,0 +1,452 @@
+package rendezvous
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	"database/sql"
+	"encoding/binary"
+	"fmt"
+	"os"
+	"time"
+
+	_ "github.com/mattn/go-sqlite3"
+
+	peer "github.com/libp2p/go-libp2p-peer"
+)
+
+type DB struct {
+	db *sql.DB
```

Comment:
i'm happy to have this be sqlite as long as its just an implementation detail (anyone could implement the service using any other form of backing datastore)

Reply:
Yeah, it's totally a service implementation detail.

Reply:
After the restructuring in https://github.com/libp2p/go-libp2p-rendezvous/commit/c487c2077b1528816577f043c1f56d143915a63b the implementation of the database is hidden behind an opaque interface.

So we can have implementations using different datastores etc; up to the user to choose the database backing the service.